### PR TITLE
fix super evaluation order tests

### PR DESCRIPTION
**Pull request by devsnek**

### CODE REVIEW

**Comment by ljharb** [comment]
Is it that the test I added was correct, but I forgot to remove/update an existing test - and now your PR removes my added test in favor of updating the existing one?

**Comment by devsnek** [comment]
@ljharb yep

File: test/language/expressions/super/call-proto-not-ctor.js [hunk 1]
```diff
@@ -3,43 +3,29 @@
...
 var evaluatedArg = false;
-var caught;
 class C extends Object {
   constructor() {
-    try {
-      super(evaluatedArg = true);
-    } catch (err) {
-      caught = err;
-    }
+    super(evaluatedArg = true);
```
Comment (leobalter) [inline test/language/expressions/super/call-proto-not-ctor.js]: it seems interesting to preserve the try catch so we can observe the abrupt completion also happening here in the constructor.

File: test/language/expressions/super/call-proto-not-ctor.js [hunk 2]
```diff
@@ -3,43 +3,29 @@
 /*---
 esid: sec-super-keyword
 es6id: 12.3.5
```
Comment (leobalter) [inline test/language/expressions/super/call-proto-not-ctor.js]: please remove the es6id tag, this is not valid for es6 anymore.

File: test/language/expressions/class/super-evaluation-order.js [hunk 3]
```diff
@@ -1,29 +0,0 @@
...
-  [...]
-  1. Let _func_ be ! GetSuperConstructor().
-  2. Let _argList_ be ? ArgumentListEvaluation of |Arguments|.
-  3. If IsConstructor(_func_) is *false*, throw a *TypeError* exception.
-  [...]
-features: [default-parameters]
----*/
-
-var x = 0;
-class C extends Object {
```
Comment (ljharb) [inline test/language/expressions/class/super-evaluation-order.js]: i made a very explicit point of extending "not null" here, so the test wouldn't start breaking when we eventually support `extends `null`.

File: test/language/statements/class/subclass/class-definition-null-proto-super.js [hunk 4]
```diff
@@ -45,7 +39,7 @@ var reachable = 0;
 class C extends null {
```
Comment (ljharb) [inline test/language/statements/class/subclass/class-definition-null-proto-super.js]: can this extend Object instead?

Comment (devsnek) [reply test/language/statements/class/subclass/class-definition-null-proto-super.js]: this test exists to test extending null, the other one is for the generic "not a constructor" case.

Comment (ljharb) [reply test/language/statements/class/subclass/class-definition-null-proto-super.js]: ah, i see - it was collapsed in the UI. thanks :-)

**Pull request merged**